### ISSUE TITLE
[Bug]: cli-kit React dependency version mismatch

### ISSUE THREAD

**Issue by nboliver-ventureweb** [Shopify/cli#3800]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

cli-kit and its deps should have matching React versions 

### Actual behavior

cli-kit has React pinned at v18.2.0
ink has it listed as ^18.0.0
This is causing ink to install v18.3.0 and cli-kit to install 18.2.0

<img width="614" alt="image" src="https://github.com/Shopify/cli/assets/132302189/48227062-b8c2-4d8d-a324-a81772891c45">

### Verbose output

```shell
n/a
```

### Reproduction steps

1. Install @shopify/cli-kit
2. Run `npm list react`
3. See different React versions
5. Run a command and get the `You might have more than one copy of React in the same app` error

### Operating System

Mac OS

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.59.1

**Comment by imrostom** [comment]
How y solved this issue ?

**Comment by nboliver-ventureweb** [comment]
Yes having two versions is affecting the project. I have used `overrides` in package.json to fix it for now. 
I have Shopify CLI installed globally with Homebrew, but this project is using Shopify CLI installed local to the project to ensure everyone is on the same version. Commands are being run with `npx`.
